### DISCUSSION TITLE
Full and log backup setup

### DISCUSSION THREAD

**Discussion by vpt1103** [olahallengren/sql-server-maintenance-solution#678] [Questions & Answers]
Hi @olahallengren 

We have a situation where we have an Availability Group setup between two servers. Primary and Secondary databases are configured on each of the servers within the AG. We are dropping the backups into two specific blob containers specific to the two servers respectively.  We have utilized your script for the Maintenance backup solution for each of the servers to do a FULL and LOG backup separately. 

When the FULL backup job runs on server 1, it moves the *.bak files of the Primary databases to container specified for server 1, and when the FULL backup job runs on server 2 it moves the *.bak files of the Primary databases to container specified for server 2.

Our issue is with the LOG backup job, when it runs on server1, it moves the *.trn of the Secondary database to container specified for server 1 and the same happens on server 2.  Our setup for the backup jobs are as follows:

Full backup job on server 1 and server 2

EXECUTE [dbo].[DatabaseBackup]
@Databases = 'AVAILABILITY_GROUP_DATABASES',
@URL = '<cannot disclose for security reasons>',
@Credential = '<cannot disclose for security reasons>',
@BackupType = 'FULL',
@Verify = 'Y',
@CheckSum = 'Y',
@LogToTable = 'Y'

Log backup job on server 1 and server 2

EXECUTE [dbo].[DatabaseBackup]
@Databases = 'AVAILABILITY_GROUP_DATABASES',
@URL = '<cannot disclose for security reasons>'',
@Credential = '<cannot disclose for security reasons>',
@BackupType = 'LOG',
@Verify = 'Y',
@CheckSum = 'Y',
@LogToTable = 'Y'

Could you help us to understand what we are missing here and fix this? 

We would like to have the LOG backup *.trn of Server 1 into container 1 and the same with Server 2.

Kind regards

**Comment by ghost** [comment]
Look at the Backup Preferences on your Availability groups. If you AGs will failover at any time, so will LOG backup
